app/models/movie.rb

```diff
@@ -0,0 +1,14 @@
+class Movie < ApplicationRecord
+  before_create :set_available_inventory
```

Comment:
Good use of a model filter here.

Another strategy is, instead of storing available inventory as a column in the DB, compute it on the fly based on what rentals currently exist for this movie. This would be a little more work, but avoids storing redundant information.